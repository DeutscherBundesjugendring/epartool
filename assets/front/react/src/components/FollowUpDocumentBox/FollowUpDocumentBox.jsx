import React from 'react';
import PropTypes from 'prop-types';
import moment from 'moment';

const FollowDocumentBox = ({
  author,
  date,
  dateMonthYearOnly,
  description,
  previewImageLink,
  title,
  type,
  typeActionLabel,
  typeEndLabel,
  typeRejectedLabel,
  typeSupportingLabel,
}) => {
  let glypClasses = 'followup-type-icon glyphicon';
  let isType = false;
  let glypTitle = '';

  if (type === 'supporting') {
    glypClasses += ' glyphicon-heart';
    glypTitle = typeSupportingLabel;
    isType = true;
  } else if (type === 'action') {
    glypClasses += ' glyphicon-play';
    glypTitle = typeActionLabel;
    isType = true;
  } else if (type === 'rejected') {
    glypClasses += ' glyphicon-minus-sign';
    glypTitle = typeRejectedLabel;
    isType = true;
  } else if (type === 'end') {
    glypClasses += ' glyphicon-lock';
    glypTitle = typeEndLabel;
    isType = true;
  }

  return (
    <div
      className="
        well
        well-bordered
        followup-flow
        followup-well
        followup-well-collapsible
        followup-well-link
        followup-type-wrap
        followup-type-wrap-inner
      "
    >
      <div className="js-followup-box-head">
        {isType && (
          <div className="followup-type followup-type-right followup-type-right-alt">
            <span className={glypClasses} aria-hidden="true" />
            <span className="followup-type-title">{glypTitle}</span>
          </div>
        )}
        <h2 className="h4">{title}</h2>
        <p className="text-muted">
          {dateMonthYearOnly
            ? moment(date).format('MMMM YYYY')
            : moment(date).format('D MMMM YYYY')}
          <span className="offset-left offset-right">|</span>
          {author}
        </p>
        <img
          src={previewImageLink}
          alt={title}
          width="120"
          className="offset-bottom img-responsive"
        />
      </div>
      <div
        className="js-followup-box-content"
        // eslint-disable-next-line react/no-danger
        dangerouslySetInnerHTML={{ __html: description }}
      />
    </div>
  );
};

FollowDocumentBox.propTypes = {
  author: PropTypes.string.isRequired,
  date: PropTypes.objectOf(Date).isRequired,
  dateMonthYearOnly: PropTypes.bool.isRequired,
  description: PropTypes.string.isRequired,
  previewImageLink: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  type: PropTypes.oneOf(['general', 'supporting', 'action', 'rejected', 'end']).isRequired,
  typeActionLabel: PropTypes.string.isRequired,
  typeEndLabel: PropTypes.string.isRequired,
  typeRejectedLabel: PropTypes.string.isRequired,
  typeSupportingLabel: PropTypes.string.isRequired,
};

export default FollowDocumentBox;
