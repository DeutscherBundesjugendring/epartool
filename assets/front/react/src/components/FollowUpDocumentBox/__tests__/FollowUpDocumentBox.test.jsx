import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import React from 'react';
import injectTapEventPlugin from 'react-tap-event-plugin';
import FollowUpDocumentBox from '../FollowUpDocumentBox';


injectTapEventPlugin();

const types = ['general', 'supporting', 'action', 'rejected', 'end'];
const getElement = type => (
  <FollowUpDocumentBox
    type={type}
    title="Document title"
    author="Author of document"
    description="Description of document"
    date={new Date('2017-01-01')}
    dateMonthYearOnly={false}
    previewImageLink="http://www.example.com/image.jpg"
    typeActionLabel="typeAction"
    typeEndLabel="typeEnd"
    typeRejectedLabel="typeRejected"
    typeSupportingLabel="typeSupporting"
  />
);


describe('rendering', () => {
  types.forEach((type) => {
    it('renders correctly with long date', () => {
      const tree = shallow(
        React.cloneElement(getElement(type))
      );

      expect(shallowToJson(tree)).toMatchSnapshot();
    });
  });

  it('renders correctly with short date', () => {
    const tree = shallow(
      React.cloneElement(getElement(types[0]), { dateMonthYearOnly: true })
    );

    expect(shallowToJson(tree)).toMatchSnapshot();
  });
});
