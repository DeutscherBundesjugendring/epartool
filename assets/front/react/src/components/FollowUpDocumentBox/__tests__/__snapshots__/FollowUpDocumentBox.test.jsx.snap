// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering renders correctly with long date 1`] = `
<div
  className=" well well-bordered followup-flow followup-well followup-well-link followup-type-wrap followup-type-wrap-inner "
>
  <div
    className="followup-type followup-type-right followup-type-right-alt"
  >
    <span
      aria-hidden="true"
      className="followup-type-icon glyphicon glyphicon-lock"
    />
    <span
      className="followup-type-title"
    />
  </div>
  <p>
    Document title
  </p>
  <p>
    Author of document
  </p>
  <p>
    1 January 2017
  </p>
  <img
    alt="Document title"
    className="offset-bottom img-responsive"
    src="http://www.example.com/image.jpg"
    width="80"
  />
  <div
    className="offset-bottom"
    dangerouslySetInnerHTML={
      Object {
        "__html": "Description of document",
      }
    }
  />
  <RaisedButton
    disabled={false}
    label="Download document"
    onTouchTap={[Function]}
  />
</div>
`;

exports[`rendering renders correctly with long date 2`] = `
<div
  className=" well well-bordered followup-flow followup-well followup-well-link followup-type-wrap followup-type-wrap-inner "
>
  <div
    className="followup-type followup-type-right followup-type-right-alt"
  >
    <span
      aria-hidden="true"
      className="followup-type-icon glyphicon glyphicon-heart"
    />
    <span
      className="followup-type-title"
    >
      Supporting
    </span>
  </div>
  <p>
    Document title
  </p>
  <p>
    Author of document
  </p>
  <p>
    1 January 2017
  </p>
  <img
    alt="Document title"
    className="offset-bottom img-responsive"
    src="http://www.example.com/image.jpg"
    width="80"
  />
  <div
    className="offset-bottom"
    dangerouslySetInnerHTML={
      Object {
        "__html": "Description of document",
      }
    }
  />
  <RaisedButton
    disabled={false}
    label="Download document"
    onTouchTap={[Function]}
  />
</div>
`;

exports[`rendering renders correctly with long date 3`] = `
<div
  className=" well well-bordered followup-flow followup-well followup-well-link followup-type-wrap followup-type-wrap-inner "
>
  <div
    className="followup-type followup-type-right followup-type-right-alt"
  >
    <span
      aria-hidden="true"
      className="followup-type-icon glyphicon glyphicon-play"
    />
    <span
      className="followup-type-title"
    >
      Action
    </span>
  </div>
  <p>
    Document title
  </p>
  <p>
    Author of document
  </p>
  <p>
    1 January 2017
  </p>
  <img
    alt="Document title"
    className="offset-bottom img-responsive"
    src="http://www.example.com/image.jpg"
    width="80"
  />
  <div
    className="offset-bottom"
    dangerouslySetInnerHTML={
      Object {
        "__html": "Description of document",
      }
    }
  />
  <RaisedButton
    disabled={false}
    label="Download document"
    onTouchTap={[Function]}
  />
</div>
`;

exports[`rendering renders correctly with long date 4`] = `
<div
  className=" well well-bordered followup-flow followup-well followup-well-link followup-type-wrap followup-type-wrap-inner "
>
  <div
    className="followup-type followup-type-right followup-type-right-alt"
  >
    <span
      aria-hidden="true"
      className="followup-type-icon glyphicon glyphicon-minus-sign"
    />
    <span
      className="followup-type-title"
    >
      Rejected
    </span>
  </div>
  <p>
    Document title
  </p>
  <p>
    Author of document
  </p>
  <p>
    1 January 2017
  </p>
  <img
    alt="Document title"
    className="offset-bottom img-responsive"
    src="http://www.example.com/image.jpg"
    width="80"
  />
  <div
    className="offset-bottom"
    dangerouslySetInnerHTML={
      Object {
        "__html": "Description of document",
      }
    }
  />
  <RaisedButton
    disabled={false}
    label="Download document"
    onTouchTap={[Function]}
  />
</div>
`;

exports[`rendering renders correctly with long date 5`] = `
<div
  className=" well well-bordered followup-flow followup-well followup-well-link followup-type-wrap followup-type-wrap-inner "
>
  <div
    className="followup-type followup-type-right followup-type-right-alt"
  >
    <span
      aria-hidden="true"
      className="followup-type-icon glyphicon glyphicon-lock"
    />
    <span
      className="followup-type-title"
    >
      End
    </span>
  </div>
  <p>
    Document title
  </p>
  <p>
    Author of document
  </p>
  <p>
    1 January 2017
  </p>
  <img
    alt="Document title"
    className="offset-bottom img-responsive"
    src="http://www.example.com/image.jpg"
    width="80"
  />
  <div
    className="offset-bottom"
    dangerouslySetInnerHTML={
      Object {
        "__html": "Description of document",
      }
    }
  />
  <RaisedButton
    disabled={false}
    label="Download document"
    onTouchTap={[Function]}
  />
</div>
`;

exports[`rendering renders correctly with short date 1`] = `
<div
  className=" well well-bordered followup-flow followup-well followup-well-link followup-type-wrap followup-type-wrap-inner "
>
  <p>
    Document title
  </p>
  <p>
    Author of document
  </p>
  <p>
    January 2017
  </p>
  <img
    alt="Document title"
    className="offset-bottom img-responsive"
    src="http://www.example.com/image.jpg"
    width="80"
  />
  <div
    className="offset-bottom"
    dangerouslySetInnerHTML={
      Object {
        "__html": "Description of document",
      }
    }
  />
  <RaisedButton
    disabled={false}
    label="Download document"
    onTouchTap={[Function]}
  />
</div>
`;
